src/reducers/todosReducer.js

```diff
@@ -1,34 +1,58 @@
-import actionConstants from '../constants/action_constants'
-import toDoReducer from './toDoReducer'
+import actionConstants from '../constants/action_constants';
 
 function getId(todos){
-  return todos.reduce((maxId, todo) => {
-    return Math.max(todo.id, maxId)
-  }, -1) + 1
+  return todos.ids.reduce((maxId, id) => {
+    return Math.max(id, maxId);
+  }, -1) + 1;
 }
 
-export default function toDosReducer(todos = [{id: 1, text: "Foo", completed: false}], action){
+/*
+  Im deliberatly doing this differently to show you a different approach.
+  Further reading http://redux.js.org/docs/advanced/AsyncActions.html#note-on-nested-entities
+  P.s. If you use immutablity this will be slightly different
+*/
+
+const initialState = {
+  ids: [1],
+  data: {
+    1: {
+      id: 1,
+      text: "Foo",
+      completed: false
+    }
+  }
+};
+
+
+export default function toDosReducer(todos = initialState, action){
 
   switch(action.type){
     case actionConstants.ADD_TODO:
-      return [{
-        id: getId(todos),
+      var id = getId(todos);
+
+      todos.ids.push(id);
+      todos.data[id] = {
+        id,
         text: action.data.text,
         completed: false
-      }, ...todos]
+      };
+
+      return todos;
 
     case actionConstants.TOGGLE_TODO:
-      return todos.map(toDo => {
-        return toDo.id === action.data.id ? toDoReducer(toDo, action) : toDo
-      })
+      var todo = todos.data[action.data.id];
+      todo.completed = !todo.completed;
+
+      return todos;
 
     case actionConstants.DELETE_TODO:
```

Comment:
Hmmm interesting that it doesnt

http://stackoverflow.com/a/37777800

Interesting approach to an alternative